moveit_core/collision_detection_fcl/src/collision_common.cpp

```diff
@@ -517,6 +517,13 @@ bool distanceCallback(fcl::CollisionObjectd* o1, fcl::CollisionObjectd* o2, void
   }
 
   fcl_result.min_distance = dist_threshold;
+  if ((o1->getObjectType() == fcl::OT_OCTREE &&
+       !std::static_pointer_cast<const fcl::OcTreed>(o1->collisionGeometry())->getRoot()) ||
+      (o1->getObjectType() == fcl::OT_OCTREE &&
+       !std::static_pointer_cast<const fcl::OcTreed>(o1->collisionGeometry())->getRoot()))
```

Comment:
I guess the second check should be about `o2`, right?
Also, please add a comment explaining that `fcl::distance` misbehaves for this in the version you use.

Reply:
Ah yes, I copy-pasted that and failed to change o1 to o2. Done.